lib/internal/url.js

```diff
@@ -219,6 +219,14 @@ class URL {
     if (base !== undefined && !(base instanceof URL))
       base = new URL(String(base));
     parse(this, input, base);
+
+    Object.defineProperty(this, Symbol.toStringTag, {
+      value: 'URL',
+      writable: false,
+      enumerable: false,
+      configurable: true
+    });
```

Comment:
@TimothyGu that would still work when defining a getter on the prototype but without causing an extra slot allocation for every single URL/URLSearchParams object.

This is not that big of a deal, but a user might have hundreds of thousands of URL objects at once and not allocating the extra memory would be nice.